packages/astro/src/core/render/core.ts

```diff
@@ -140,11 +140,10 @@ export async function render(opts: RenderOptions): Promise<Response> {
 		ssr,
 	});
 
-	if (!Component.isAstroComponentFactory) {
-		const props: Record<string, any> = { ...(pageProps ?? {}), 'server:root': true };
-		const html = await renderComponent(result, Component.name, Component, props, null);
-		return new Response(html.toString(), result.response);
-	} else {
-		return await renderPage(result, Component, pageProps, null);
+	// Support `export const components` for `MDX` pages
+	if (typeof (mod as any).components === 'object') {
+		Object.assign(pageProps, { components: (mod as any).components });
 	}
+
+	return await renderPage(result, Component, pageProps, null);
```

Comment:
Will this code path never hit for a component factory now?

Reply:
`renderPage` now handles component factories (`.astro` pages) and runs anything else through `renderComponent` (to support `.mdx` pages)